packages/knip/src/plugins/ladle/index.ts

```diff
@@ -0,0 +1,40 @@
+import type { EnablerPatterns } from '#p/types/config.js';
+import type { IsPluginEnabled, Plugin, ResolveEntryPaths } from '#p/types/plugins.js';
+import { hasDependency } from '#p/util/plugin.js';
+import type { LadleConfig } from './types.js';
+import { toEntryPattern } from '../../util/protocols.js';
+
+// https://ladle.dev/docs/config
+
+const title = 'ladle';
+
+const enablers: EnablerPatterns = [/^@ladle\//];
+
+const isEnabled: IsPluginEnabled = ({ dependencies }) => hasDependency(dependencies, enablers);
+
+const config = ['.ladle/config.{mjs,js,ts}'];
+
+const stories: string[] = ['**/*.@(stories.@(mdx|js|jsx|mjs|ts|tsx))'];
+const restEntry: string[] = ['.ladle/components.{js,jsx,ts,tsx}', 'head.html'];
+const entry: string[] = [...restEntry, ...stories];
+
+const project = ['.ladle/**/*.{js,jsx,ts,tsx}'];
+
+const resolveEntryPaths: ResolveEntryPaths<LadleConfig> = async localConfig => {
+  const localStories = typeof localConfig.stories === 'string' ? [localConfig.stories] : localConfig.stories;
+  const localViteConfig = localConfig.viteConfig ? [localConfig.viteConfig] : [];
```

Comment:
The Vite config should be handled by the Vite plugin, so I think we can just ignore it here? Btw it would need be a `config` file, not an `entry` file.

Reply:
For ladle it might be necessary to have a dedicated / different viteConfig file. So in my case I have `~/.vite.config.ts` and for ladle `~/.ladle/vite.config.ts`. Only if you don't use the default one, this parameter is set and then should be respected. Otherwise, I get it listed as "unused file".

Reply:
OK in that case I think it's interesting to look into using the `resolveConfig` function of the Vitest plugin (which is basically the same as the Vite plugin).

Simplified inside `resolveConfig`: `const viteDeps = resolveConfig(await load(localConfig.viteConfig))`

Does that make sense?

Reply:
Changed 👍 I wasn't aware that it is allowed to use other plugins.

Reply:
… I cannot get rid of the message that the file is unused, stuck here.

```
Unused files (1)
.ladle/vite.config.ts
```

I also tried things like 

```
const resolveConfig: ResolveConfig<LadleConfig> = async (localConfig, options) =>
  resolveVitestConfig(localConfig.viteConfig ? await load(join(process.cwd(), localConfig.viteConfig)) : [], options);
```

Maybe someone with more insights can help here? 😇

Reply:
Perhaps confusingly, loading the file does not "use" it (because it's not part of the dependency graph). For this you could add it as an entry file (contrary to what I said earlier, sorry, but that was for another reason).

Reply:
I re-added it additionally as an entry file 👍